ballerina/graphql_client.bal

```diff
@@ -0,0 +1,103 @@
+        http:Client|http:ClientError httpClient = new (serviceUrl, clientConfig);
+        if httpClient is http:ClientError {
+             return error RequestError("GraphQL Client Error", httpClient);
+        }
+        self.httpClient = httpClient;
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data and extensions 
+    # which is a subtype of GenericResponse.
+    #
+    # + document - The GraphQL query or mutation. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
+    # + variables - The GraphQL variables. For example `{"code": "<variable_value>"}`.
+    # + headers - The GraphQL API headers to execute each query 
+    # + targetType - The payload, which is expected to be returned after data binding. For example 
+    #                `type CountryByCodeResponse record {| map<json?> __extensions?; record {| record{|string name;|}? country; |} data;`
+    # + return - The GraphQL response or a `graphql:ClientError` if failed to execute the query
+    remote isolated function executeWithType(string document, map<anydata>? variables = (), 
+                                             map<string|string[]>? headers = (), 
+                                             typedesc<GenericResponse|record{}|json> targetType = <>) 
+                                             returns targetType|ClientError = @java:Method {
+        'class: "io.ballerina.stdlib.graphql.runtime.client.QueryExecutor",
+        name: "executeWithType"
+    } external;
+
+    private isolated function processExecuteWithType(typedesc<GenericResponse|record{}|json> targetType, 
+                                                     string document, map<anydata>? variables, 
+                                                     map<string|string[]>? headers) 
+                                                     returns GenericResponse|record{}|json|ClientError {
+        http:Request request = new;
+        json graphqlPayload = getGraphqlPayload(document, variables);
+        request.setPayload(graphqlPayload);
+        json|http:ClientError httpResponse = self.httpClient->post("", request, headers = headers);
+
+        if httpResponse is http:ClientError {
+            return handleHttpClientErrorResponse(httpResponse);
+        } 
+        GenericResponseWithErrors|record{}|json response = check performDataBinding(targetType, httpResponse);
+        map<json> responseMap = <map<json>> httpResponse;
+        if responseMap.hasKey("errors") {
+            return handleGraphqlErrorResponse(responseMap);
+        } 
+        return response;
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data, extensions and errors 
+    # which is a subtype of GenericResponseWithErrors.
+    #
+    # + document - The GraphQL query or mutation. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
```

Comment:
Ditto.

Reply:
The `operationName` is added as a new optional parameter in the client functions & test cases were added in [f26678c](https://github.com/ballerina-platform/module-ballerina-graphql/pull/663/commits/f26678c4ceef61a6ce293851142e167fd6e26c07). Thank you.